components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/OAuth2ScopeService.java

```diff
@@ -173,6 +207,42 @@ public boolean isScopeExists(String name) throws IdentityOAuth2ScopeException {
         return isScopeExists;
     }
 
+    /**
+     * Check the existence of a scope depends on scope type. Type can be OAUTH2 scopes or OIDC scopes.
+     *
+     * @param name              Name of the scope.
+     * @param includeOIDCScopes Include OIDC scopes as well.
+     * @return True if scope with the given scope name exists.
+     * @throws IdentityOAuth2ScopeException
+     */
+    public boolean isScopeExists(String name, boolean includeOIDCScopes) throws IdentityOAuth2ScopeException {
```

Comment:
includeOIDCScopes is not required.

Reply:
As per the offline discussion, I'll keep as it is since we don't expose OIDC scopes, via isScopeExists method check.